### PR TITLE
E1509 Refactoring SignUpController and SignUpSheetController

### PR DESCRIPTION
**Pull request by jli53**
Here are our modifications for OSS project topic E1509

### CODE REVIEW

**Comment by yangsong8** [comment]
Hi, did you make any change on the DB? Thanks.

**Comment by jli53** [comment]
Hi Yang,

We did not change anything on DB. Is there anything wrong about it?

Thanks,
Jiaming

**Comment by yangsong8** [comment]
It is about the changes on db schema. If you did not change the db, we can
ignore the changed on that file when we merge your project.

**Comment by jli53** [comment]
I checked the schematic.rb again. We did not mean to change anything on
this file. But it seems that when we delete somethhing from a file in
"view" folder, this chema file also get changed.

**Comment by yangsong8** [comment]
I mean the file: db/schema.rb. Your team committed this file and changed several hundreds on lines, which made this pull request contains more changed lines than we expected. This is the reason I asked if you guys made any change on the DB. If your answer is "no", this means the db/schema.rb is changed by Rails. So we will not worry about this file when we merge your changes.

By the way, in your final project, you do not need to commit db/schema.rb unless you really change the DB.

Thank you for your time.

**Comment by jli53** [comment]
Our answer is "no". We didn't change anything on that file. Sorry for the
trouble.

**Comment by yangsong8** [comment]
Cool, thanks.

**Comment by yangsong8** [comment]
- [ ] This project does not include any db change, we do not need to merge db/schema.rb.

File: app/models/participant.rb [hunk 1]
```diff
@@ -168,4 +168,8 @@ def get_scores(questions)
 
       scores
     end
+
+    def self.find_participant_byassign(user_id,assign_id)
```
Comment (yangsong8) [inline app/models/participant.rb]: You should also check if the participant.type is "AssignmentParticipant".

Comment (jli53) [inline app/models/participant.rb]: We rewrite this function according to the original code "participant =
Participant.where(user_id: session[:user].id, parent_id:
assignment_id).first".
We did not check if the original code miss some important checks. I think
for this part, our task is just  to improve format and style.

**Pull request closed without merging**